### PR TITLE
Adding bceid lib

### PR DESCRIPTION
**Pull request by alexjoybc**
Adding bceid starter

### CODE REVIEW

File: src/spring-bceid-starter/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# spring-bceid-starter
+
+A stater to facilitate usage of bceid service.
```
Comment (sdevalapurkar-bcgov) [inline src/spring-bceid-starter/README.md:3]:
```suggestion
A starter to facilitate usage of the BCeID service.
```

File: src/spring-bceid-starter/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+| [bcgov.bceid.service.uri](#cgovbceidserviceuri) | String | Yes |
+| [bcgov.bceid.service.username](#bcgovbceidserviceusername) | String | Yes |
+| [bcgov.bceid.service.password](#bcgovbceidservicepassword) | String | Yes |
+| [bcgov.bceid.service.onlineServiceId](#bcgovbceidserviceonlineServiceId) | String | Yes |
+
+#### bcgov.bceid.service.uri
+
+* Value type is String
+
+Sets the bceid service URI
```
Comment (sdevalapurkar-bcgov) [inline src/spring-bceid-starter/README.md:32]:
```suggestion
Sets the BCeID service URI
```

File: src/spring-bceid-starter/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+#### bcgov.bceid.service.username
+
+* Value type is String
+
+Sets the username used to set basic authentication on bceid service
```
Comment (sdevalapurkar-bcgov) [inline src/spring-bceid-starter/README.md:38]:
```suggestion
Sets the username used to set basic authentication on the BCeID service
```

File: src/spring-bceid-starter/README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+#### bcgov.bceid.service.password
+
+* Value type is String
+
+Sets the password used to set basic authentication on bceid service
```
Comment (sdevalapurkar-bcgov) [inline src/spring-bceid-starter/README.md:44]:
```suggestion
Sets the password used to set basic authentication on the BCeID service
```

File: src/spring-bceid-starter/pom.xml [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+
+        </plugins>
+
+    </build>
+
+</project>
```
Comment (sdevalapurkar-bcgov) [inline src/spring-bceid-starter/pom.xml]: missing newline (nit)

Comment (TayGov) [inline src/spring-bceid-starter/pom.xml:148]: New line?

File: src/spring-bceid-starter/src/main/java/ca/bc/gov/open/bceid/starter/AutoConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@EnableConfigurationProperties(BCeIdProperties.class)
+public class AutoConfiguration {
+
+    private final BCeIdProperties bCeIdProperties;
+
+    public AutoConfiguration(BCeIdProperties bCeIdProperties) {
+        this.bCeIdProperties = bCeIdProperties;
```
Comment (sdevalapurkar-bcgov) [inline src/spring-bceid-starter/src/main/java/ca/bc/gov/open/bceid/starter/AutoConfiguration.java:23]: question:
why not just `bceidProperties`? seems like weird casing with `bCeId`

File: src/spring-bceid-starter/src/main/java/ca/bc/gov/open/bceid/starter/account/GetAccountRequest.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+    public BCeIDAccountTypeCode getRequesterBCeIDAccountTypeCode() {
+        return requesterBCeIDAccountTypeCode;
+    }
+
+    public static GetAccountRequest BusinessSelfRequest(String id) {
+        return new GetAccountRequest(id, BCeIDAccountTypeCode.BUSINESS);
+    }
+
+    public static GetAccountRequest EdsSelfRequest(String id) {
```
Comment (sdevalapurkar-bcgov) [inline src/spring-bceid-starter/src/main/java/ca/bc/gov/open/bceid/starter/account/GetAccountRequest.java:41]: comment mentioning what EDS is?

File: src/spring-bceid-starter/src/main/java/ca/bc/gov/open/bceid/starter/account/GetAccountRequest.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+    public static GetAccountRequest IndividualSelfRequest(String id) {
+        return new GetAccountRequest(id, BCeIDAccountTypeCode.INDIVIDUAL);
+    }
+
+    public static GetAccountRequest InternalSelfRequest(String id) {
+        return new GetAccountRequest(id, BCeIDAccountTypeCode.INTERNAL);
+    }
+
+    public static GetAccountRequest LdbSelfRequest(String id) {
```
Comment (sdevalapurkar-bcgov) [inline src/spring-bceid-starter/src/main/java/ca/bc/gov/open/bceid/starter/account/GetAccountRequest.java:53]: same here, comment about LDB?

File: src/spring-bceid-starter/src/main/java/ca/bc/gov/open/bceid/starter/account/mappers/NameMapper.java [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+@Mapper(uses = BCeIdMapper.class)
+public interface NameMapper {
+
+    @Mapping(target = "firstName", source = "firstname", qualifiedByName=  { "BCeIdMapper", "toString"} )
+    @Mapping(target = "middleName", source = "middleName", qualifiedByName=  { "BCeIdMapper", "toString"} )
+    @Mapping(target = "otherMiddleName", source = "otherMiddleName", qualifiedByName=  { "BCeIdMapper", "toString"} )
+    @Mapping(target = "surname", source = "surname", qualifiedByName=  { "BCeIdMapper", "toString"} )
```
Comment (sdevalapurkar-bcgov) [inline src/spring-bceid-starter/src/main/java/ca/bc/gov/open/bceid/starter/account/mappers/NameMapper.java:14]: interesting, I thought it was called `lastName` not `surname`

File: src/spring-bceid-starter/src/main/resources/META-INF/spring.factories [hunk 10]
```diff
@@ -0,0 +1,3 @@
+org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
+  ca.bc.gov.open.bceid.starter.AutoConfiguration
+
```
Comment (TayGov) [inline src/spring-bceid-starter/src/main/resources/META-INF/spring.factories:3]: Is there empty space on this new line?

File: src/spring-starters-bom/pom.xml [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+                <version>${joda-time.version}</version>
+            </dependency>
+
+
+        </dependencies>
+    </dependencyManagement>
+
+
+
+</project>
```
Comment (sdevalapurkar-bcgov) [inline src/spring-starters-bom/pom.xml]: newline

File: src/spring-bceid-starter/pom.xml (hunk 5 above)
Comment (akroon3r) [reply src/spring-bceid-starter/pom.xml]: :shipit:

File: src/spring-bceid-starter/src/main/java/ca/bc/gov/open/bceid/starter/AutoConfiguration.java (hunk 6 above)
Comment (alexjoybc) [reply src/spring-bceid-starter/src/main/java/ca/bc/gov/open/bceid/starter/AutoConfiguration.java:23]: good call however all the services are BCeID, so I kept it consistent...

File: src/spring-bceid-starter/src/main/java/ca/bc/gov/open/bceid/starter/account/GetAccountRequest.java (hunk 7 above)
Comment (alexjoybc) [reply src/spring-bceid-starter/src/main/java/ca/bc/gov/open/bceid/starter/account/GetAccountRequest.java:46]: no idea. it's just supporting that type

File: src/spring-bceid-starter/src/main/java/ca/bc/gov/open/bceid/starter/account/GetAccountRequest.java (hunk 8 above)
Comment (alexjoybc) [reply src/spring-bceid-starter/src/main/java/ca/bc/gov/open/bceid/starter/account/GetAccountRequest.java:73]: same here

File: src/spring-bceid-starter/src/main/java/ca/bc/gov/open/bceid/starter/account/mappers/NameMapper.java (hunk 9 above)
Comment (alexjoybc) [reply src/spring-bceid-starter/src/main/java/ca/bc/gov/open/bceid/starter/account/mappers/NameMapper.java:14]: that's when it's going out to BCeId, it's called surname....

File: src/spring-bceid-starter/README.md (hunk 2 above)
Comment (alexjoybc) [inline src/spring-bceid-starter/README.md:32]:
```suggestion
Sets the BCeID service URI
```

File: src/spring-bceid-starter/README.md (hunk 3 above)
Comment (alexjoybc) [inline src/spring-bceid-starter/README.md:38]:
```suggestion
Sets the username used to set basic authentication on the BCeID service
```

File: src/spring-bceid-starter/README.md (hunk 4 above)
Comment (alexjoybc) [inline src/spring-bceid-starter/README.md:44]:
```suggestion
Sets the password used to set basic authentication on the BCeID service
```

**Pull request merged**